### PR TITLE
sg: Provide better error message if command does not start up or then fails

### PR DESCRIPTION
**Pull request by mrnugget**
Previously only the "failed to rebuild" message was nice. Now the "failed even install it" message is formatted too.

image

Yes, the error handling is not the nicest yet, but I think it's okay for now.

### CODE REVIEW

File: dev/sg/run.go [hunk 1]
```diff
@@ -48,15 +48,83 @@ func run(ctx context.Context, cmds ...Command) error {
...
+		cmdOut = e.output
+	default:
+		message = fmt.Sprintf("Failed to run %s: %s", cmdName, err)
+	}
+
+	separator := strings.Repeat("-", 80)
+	if cmdOut != "" {
+		line := output.Linef(
+			"", output.StyleWarning,
+			"%s\n%s%s %s%s%s: \n%s%s%s%s",
```
Comment (eseliger) [inline dev/sg/run.go:114]: Is this Christmas decoration or a format string 😅

**Review by eseliger — APPROVED** [review]

File: dev/sg/run.go [hunk 2]
```diff
@@ -48,15 +48,83 @@ func run(ctx context.Context, cmds ...Command) error {
...
 					cancel()
 				}
 			}
 		}(cmd, chs[i])
 	}
 
 	wg.Wait()
-	return <-errs
+
+	fail := <-failures
```
Comment (jeanduplessis) [inline dev/sg/run.go:60]: nitpick (non-blocking): `fail` my mind went to thinking of fail as a verb, maybe a better name is simply `failure`.

File: dev/sg/run.go [hunk 3]
```diff
@@ -48,15 +48,83 @@ func run(ctx context.Context, cmds ...Command) error {
...
+	printCmdError(out, fail.cmdName, fail.err)
+	return fail
+}
+
+// runFailed is returned by run when a command failed to run and run exits
+type runFailed struct {
```
Comment (jeanduplessis) [inline dev/sg/run.go:66]: nitpick (non-blocking): Similar to the above message I'd go with `failedRun` as a description of the struct.

**Review by jeanduplessis — APPROVED** [review]

File: dev/sg/run.go (hunk 2 above)
Comment (mrnugget) [reply dev/sg/run.go]: Done!

File: dev/sg/run.go (hunk 3 above)
Comment (mrnugget) [reply dev/sg/run.go]: Done!

**Pull request merged**